app/src/main/java/se/rebeccazadig/bokholken/login/UserViewModel.kt

```diff
@@ -1,44 +1,84 @@
 package se.rebeccazadig.bokholken.login
 
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.ViewModel
-import androidx.lifecycle.viewModelScope
+import android.database.Observable
+import android.util.Log
+import androidx.lifecycle.*
 import kotlinx.coroutines.launch
 import se.rebeccazadig.bokholken.myAdverts.UiState
 
-// internal data class UiState(
-//    /*uiState är MLD är observed i frag men all ändring sker i viewmodel*/
-//    val isDeleted: Boolean,
-//    val message: String?,
-// )
+internal data class UiStateSave(
+
+    val message: String?,
+)
+
 class UserViewModel : ViewModel() {
 
     private val loginRepo = LoginRepository.getInstance()
     private val userRepo = UserRepository.getInstance()
+    val userName = MutableLiveData("")
+    val userContact = MutableLiveData("")
+    val userCity = MutableLiveData("")
 
     val inProgress = MutableLiveData(false)
     private val _uiState = MutableLiveData(UiState(false, null))
```

Comment:
which UiState class you use here? It takes two parameters.  in line 25 UiState class constructor takes only one parameter

Reply:
Not sure what you mean. I use UiState and UiStateSave